### PR TITLE
PW-404: Referral System Functionality

### PR DESCRIPTION
**Pull request by alexnatalia**
# Task

[PW-404]: Referral system support via web app

## Changes

* Added Referral System Functionality.
* Updated lib version, doc pallet info.

[PW-404]: https://soramitsu.atlassian.net/browse/PW-404

### CODE REVIEW

File: packages/util/src/BaseApi.ts [hunk 1]
```diff
@@ -438,6 +450,12 @@ export class BaseApi {
         return this.api.tx.pswapDistribution.claimIncentive()
       case Operation.ClaimExternalRewards:
         return this.api.tx.rewards.claim('0xa8811ca9a2f65a4e21bd82a1e121f2a7f0f94006d0d4bcacf50016aef0b67765692bb7a06367365f13a521ec129c260451a682e658048729ff514e77e4cdffab1b') // signature mock
+      case Operation.ReferralReserveXor:
+        return this.api.tx.referrals.reserve('0')
+      case Operation.ReferralUnreserveXor:
+        return this.api.tx.referrals.unreserve('0')
+      case Operation.ReferralSetInvitedUser:
+        return this.api.tx.referrals.setReferrer('')
```
Comment (Nikita-Polyakov) [inline packages/util/src/BaseApi.ts:458]: Will this extrinsic work with empty account address? I mean we can insert here some mock of account address to be sure for extrinsic success

File: packages/util/src/api.ts [hunk 2]
```diff
@@ -1569,6 +1569,76 @@ export class Api extends BaseApi {
...
+  }
+
+  /**
+   * Returns invited users of the referral
+   * @param referralId address of referral account
+   * @returns array of invited users
+   */
+  public async getInvitedUsers (referralId: string): Promise<Array<string>> {
+    const referrals = await this.api.query.referrals.referrals(referralId) as any
+    if (!referrals) {
```
Comment (Nikita-Polyakov) [inline packages/util/src/api.ts:1590]: Looks like if account hasn't referrals, this method already returns empty array (checked it in polkadotjs apps). Or another falsy values could be returned by `api.query.referrals.referrals`?

File: packages/util/src/api.ts [hunk 3]
```diff
@@ -1569,6 +1569,76 @@ export class Api extends BaseApi {
...
+
+  /**
+   * Transfer XOR balance from the referral account to the special account
+   * This balance can be used by referrals to pay the fee
+   * @param amount balance to reserve
+   */
+  public async reserveXor (amount: NumberLike): Promise<void> {
+    assert(this.account, Messages.connectWallet)
+    const asset = await this.getAssetInfo(KnownAssets.get(KnownSymbols.XOR).address)
+    const assetAddress: string = asset?.address
```
Comment (Nikita-Polyakov) [inline packages/util/src/api.ts:1604]: Do we need the call above, if we can just use `KnownAssets.get(KnownSymbols.XOR)` or (`XOR` exported from assets.ts) data?

**Review by Nikita-Polyakov — APPROVED** [review]

File: packages/util/src/BaseApi.ts (hunk 1 above)
Comment (alexnatalia) [reply packages/util/src/BaseApi.ts:458]: Yes, it works without any mock address.

File: packages/util/src/api.ts (hunk 2 above)
Comment (alexnatalia) [reply packages/util/src/api.ts:1590]: Yes, you are right, fixed.

**Review by Nikita-Polyakov — APPROVED** [review]

**Review by stefashkaa — APPROVED** [review]
Please change version to 1.7.0

**Pull request merged**